python/Ganga/GPIDev/Lib/GangaList/GangaList.py

```diff
@@ -52,8 +52,8 @@ def makeGangaListByRef(_list, preparable=False):
     result = GangaList()
     if len(_list) == 0:
         return result
-    temp_list = [stripProxy(element) for element in _list]
-    result._list = temp_list
+
+    result._list.extend([stripProxy(element) for element in _list])
```

Comment:
Indeed at the moment the last example you put would fail. This is because `extend` is simply:
```python
def extend(self, ittr):
    for i in ittr:
        self.append(i)
```
It all really depends on exactly what semantics we want things to have. At the GPI-level I think we want things to be copied gratuitously in order to avoid stale references etc. In this case, we would need to make sure that `GangaList._export_extend` correctly deep-copies the objects before appending them.